test/e2e/framework/service_util.go

```diff
@@ -1193,11 +1197,9 @@ func ValidateEndpointsOrFail(c clientset.Interface, namespace, serviceName strin
 }
 
 // StartServeHostnameService creates a replication controller that serves its hostname and a service on top of it.
-func StartServeHostnameService(c clientset.Interface, internalClient internalclientset.Interface, ns, name string, port, replicas int) ([]string, string, error) {
+func StartCustomizedServeHostnameService(c clientset.Interface, internalClient internalclientset.Interface, ns, name string, port, replicas int, tweak func(svc *v1.Service)) ([]string, string, error) {
```

Comment:
you should document tweak. why not take in a Service and merge them? What is going on in tweak()

Reply:
If we take service as an parameter, does that means we need to check every single attributes to make sure we are not overwriting them?